### PR TITLE
fix: Stream settings page redesign

### PR DESCRIPTION
**Pull request by nikhilsaikethe**
This PR solves the issue #5216 #5224 

## Release Notes

- **New Features**
	- Enhanced user interface for the log stream input fields with improved layout and visibility, including dynamic width adjustments and clearer labeling.
	- Added a new column to display index size in the log streams table.
	- Introduced pagination management for schema fields, allowing for dynamic updates to the number of rows displayed.
	- Added a new prop, `isInSchema`, to enhance component layout based on schema inclusion.

- **Bug Fixes**
	- Improved error messaging for invalid data retention periods.

- **Documentation**
	- Updated language localization for clarity in schema field descriptions.

### CODE REVIEW

File: web/src/components/logstream/schema.vue [hunk 1]
```diff
@@ -416,6 +495,12 @@
 import StreamFieldsInputs from "@/components/logstream/StreamFieldInputs.vue";
 import AppTabs from "@/components/common/AppTabs.vue";
 
+import QTablePagination from "@/components/shared/grid/Pagination.vue";
+import { outlinedSchema,outlinedPerson,outlinedDelete } from "@quasar/extras/material-icons-outlined";
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/components/logstream/schema.vue:499]: yes they are available

File: web/src/components/logstream/schema.vue [hunk 2]
```diff
@@ -323,74 +375,101 @@
...
+                class="q-my-sm no-border text-bold q-mr-md"
+                padding="sm md"
+                color="primary"
+                no-caps
+                v-bind:disable="!selectedFields.length"
+                @click="updateDefinedSchemaFields"
+              >
+              <span class="flex items-center justify-start q-mr-sm">
+                <q-icon size="14px"  :name="outlinedPerson"/>
+                <q-icon size="14px" :name="outlinedSchema" />
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/components/logstream/schema.vue:413]: sure I have verified that

File: web/src/components/logstream/schema.vue [hunk 3]
```diff
@@ -238,22 +181,121 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+                    />
+                  </q-th>
+                <q-th
+                  v-for="col in props.cols"
+                  :key="col.name"
+                  :props="props"
+                >
+                <span v-if="col.icon" >
+                  <q-icon color="primary" :name="outlinedPerson"></q-icon>
+                  <q-icon color="primary" :name="outlinedSchema"></q-icon>
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/components/logstream/schema.vue:288]: yes I have verified

**Comment by gaby** [comment]
@nikhilsaikethe Since you are at it, can the dropdown to choose how many streams to see have an option for 250 or 500?

I have around 2k streams when using prometheus. When deleting them It takes around 20 tries by going 100 by 100

**Review by hengfeiyang — APPROVED** [review]

File: web/src/components/logstream/schema.vue [hunk 4]
```diff
@@ -453,13 +540,37 @@ export default defineComponent({
...
+    const isDialogOpen = ref(false);
+    const resultTotal = ref<number>(0);
+      const perPageOptions: any = [
+      { label: "5", value: 5 },
+      { label: "10", value: 10 },
+      { label: "20", value: 20 },
+      { label: "50", value: 50 },
+      { label: "100", value: 100 },
+      { label: "All", value: 0 },
+    ];
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/components/logstream/schema.vue:561]: yes it is done

File: web/src/components/logstream/schema.vue [hunk 5]
```diff
@@ -46,81 +46,62 @@
...
+                  data-test="stream-details-data-retention-input"
+                  v-model="dataRetentionDays"
+                  type="number"
+                  dense
+                  filled
+                  min="0"
+                  round
+                  class="q-mr-sm q-ml-sm data-retention-input"
+                  hide-bottom-space
+                  @change="formDirtyFlag = true"
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/components/logstream/schema.vue]: yes modified this

File: web/src/components/logstream/StreamFieldInputs.vue [hunk 6]
```diff
@@ -135,6 +138,7 @@ import { defineProps, ref } from "vue";
 import { useI18n } from "vue-i18n";
 import { outlinedDelete } from "@quasar/extras/material-icons-outlined";
 import { useStore } from "vuex";
+import { styleText } from "util";
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/components/logstream/StreamFieldInputs.vue:141]: yes I have removed this

**Pull request merged**